File: PRACTICA/Java/irrgarten/Weapon.java
package irrgarten;

public class Weapon extends CombatElement {
    public Weapon(float power, int uses) {
        super(power, uses);
    }

    public float attack() {
        return produceEffect();
    }

    @Override
    public String toString() {
        return "W" + super.toString();
    }
}

File: Ruby/Irrgarten/weapon.rb
require_relative 'combat_element'

class Weapon < Combat_element
  def initialize(power, uses)
    super(power, uses)
  end

  def attack
    return produce_effect()
  end

  def to_s
    "W" + super.to_s
  end

  public_class_method :new
end